caas/kubernetes/provider/bootstrap_test.go

```diff
@@ -54,7 +56,8 @@ func (s *bootstrapSuite) SetUpTest(c *gc.C) {
 	s.controllerUUID = "9bec388c-d264-4cde-8b29-3e675959157a"
 
 	s.controllerCfg = testing.FakeControllerConfig()
-	pcfg, err := podcfg.NewBootstrapControllerPodConfig(s.controllerCfg, controllerName, "bionic")
+	s.controllerCfg[controller.APIPort] = 17070
```

Comment:
We try and avoid using 17070 in tests. Is there a reason not to stick with 17777 set up in FakeControllerConfig() ?

Reply:
shouldn't this be a dynamic port? or we know that this is never listened on or talked to so it is arbitrary?
I agree you want to avoid 17070 and have things "accidentally" work, and ignore the config

Reply:
right, fixed!
now the expected spec is read from controllerCfg.
thanks